### PR TITLE
fix: correct query results and prevent silent data drops

### PR DESCRIPTION
**Pull request by msallin**
Correctness fixes for the v1 release. Covered by build, lint, the full mocha suite (531 passing), and an adversarial review.

## What

- **Angular aggregation.** Add a `COALESCE(value_sin_avg, SIN(value_avg))` / `COALESCE(value_cos_avg, COS(value_avg))` fallback to the tier-to-tier angular re-aggregation so a missing or flapping sin/cos column yields a correct circular mean instead of NULL or an arithmetic mean of angles. Matches the query-time fallback already used by the History API; identical to the prior result when the columns are present.
- **Ingestion drop.** Stop discarding object deltas whose only key is `timeout`: it can be a real value field, so keeping it errs away from silent data loss.
- **Default resolution.** Clamp the v1 default resolution to >= 1 ms (computed in milliseconds, mirroring the v2 provider) so a zero-duration range can't produce a 0 ms DuckDB bucket size.
- **Datetime parsing.** Normalize date-only `YYYY-MM-DD` inputs to local midnight so bare timestamps are interpreted consistently with full datetimes.
- **Spatial correlation.** On a spatial position-query error, skip correlation (return null) instead of fabricating an empty set that silently empties every correlated path; a transient error no longer blanks unrelated data.
- **Migration.** Await the legacy-retention `savePluginOptions` so the `configSchemaVersion` stamp is confirmed before start proceeds; `cancel()` now accepts the `scanning` phase.

## Scope and merge order

One of four independent v1-stabilization PRs cut from `main`. Recommended merge order: security → reliability → correctness → hygiene (trial merge of all four was conflict-free).

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -213,12 +213,24 @@ export default function (app: ServerAPI): SignalKPlugin {
...
-            `[Retention] Failed to persist legacy migration: ${(err as Error).message}`
-          );
-        }
+      // savePluginOptions is callback-based; await the promisified call so the
+      // configSchemaVersion sentinel is actually confirmed on disk within start
+      // ordering. Resolve even on error (a persistence hiccup must not block
+      // startup) but log loudly: the sentinel won't have landed, so the
+      // migration re-runs on the next restart.
+      await new Promise<void>(resolve => {
+        // currentConfig was just assigned above; the non-null assertion is
```
Comment (msallin) [inline src/index.ts]: this comment is unnecessary

File: src/HistoryAPI.ts [hunk 2]
```diff
@@ -953,10 +964,21 @@ export class HistoryAPI {
     res: Response<any, Record<string, any>>
   ) {
     try {
-      // Resolution now in SECONDS (breaking change from v0.7.0)
+      // Resolution now in SECONDS (breaking change from v0.7.0).
```
Comment (msallin) [inline src/HistoryAPI.ts]: remove   // Resolution now in SECONDS (breaking change from v0.7.0).
No history commentgs

File: src/index.ts (hunk 1 above)
Comment (msallin) [reply src/index.ts]: Removed.

File: src/HistoryAPI.ts (hunk 2 above)
Comment (msallin) [reply src/HistoryAPI.ts]: Removed the change-history line; kept only the why-comment for the auto-resolution clamp.

File: src/services/migration-service.ts [hunk 3]
```diff
@@ -440,7 +440,11 @@ export class MigrationService {
...
   cancel(jobId: string): boolean {
     const job = migrationJobs.get(jobId);
-    if (job && job.status === 'running') {
+    // Accept 'scanning' as well as 'running': the scan phase runs before the
+    // migrate loop flips the status to 'running', and cancelRequested is checked
+    // inside that loop, so a cancel issued during scanning must take effect.
+    // Matches the cancel semantics in compaction and aggregation services.
+    if (job && (job.status === 'scanning' || job.status === 'running')) {
       this.cancelRequested = true;
       return true;
```
[the remark this answers is not in the document]
Comment (msallin) [reply src/services/migration-service.ts:448]: Fixed — cancellation is now scoped per job. Added a cancelRequested flag on MigrationProgress; cancel(jobId) sets that job's flag, runMigration() checks its own job's flag, and migrate() no longer resets a shared boolean. There is no upstream concurrency guard on /api/migrate, so this prevents one cancel from stopping a different in-flight migration.

File: src/index.ts [hunk 4]
```diff
@@ -213,12 +213,22 @@ export default function (app: ServerAPI): SignalKPlugin {
...
+      // startup) but log loudly: the sentinel won't have landed, so the
+      // migration re-runs on the next restart.
+      await new Promise<void>(resolve => {
+        app.savePluginOptions(state.currentConfig!, (err?: unknown) => {
+          if (err) {
+            app.error(
+              `[Retention] Failed to persist legacy migration; the ` +
+                `configSchemaVersion stamp did not land and the migration will ` +
+                `re-run on next restart: ${(err as Error).message}`
+            );
```
[the remark this answers is not in the document]
Comment (msallin) [reply src/index.ts:229]: Fixed — now formats with `err instanceof Error ? err.message : String(err)` so a non-Error callback value is preserved in the log.

File: src/data-handler.ts [hunk 5]
```diff
@@ -660,15 +660,17 @@ function handleStreamData(
...
+      // and the worst outcome here is silently dropping archived data, so we
+      // err toward keeping an object whose only key is 'timeout'.
       const metaOnlyKeys = [
         'units',
         'meta',
         'description',
         'displayUnits',
         'zones',
-        'timeout',
       ];
```
[the remark this answers is not in the document]
Comment (msallin) [reply src/data-handler.ts:674]: Synced the copied metaOnlyKeys in tests/test-sqlite-buffer.js (removed 'timeout') so it matches handleStreamData. Note that file is a legacy standalone WebSocket script (not part of the mocha suite, which is test/**/*.test.ts), so I kept the change to the one-line sync; the canonical timeout-only assertion belongs in the mocha test suite, which is part of the deferred test-coverage work.

**Comment by motamman** [comment]
Reviewed against current main (0.7.44-beta.1): 8 of the 9 changes here are still needed — none of the 0.7.42/0.7.43 work covered them. In particular the angular re-aggregation fix (`aggregation-service.ts` — the shipped 0.7.43 smoothing work fixed query-time angular math but not the write-path `SUM(value_sin_avg …)` NULL issue), the `timeout` metaOnlyKeys drop, the spatial-error → `null` change (main already types `spatialTimestamps` as `Set | null` and null-guards the consumer, so your change is drop-in), the migration per-job cancel, and the date-only `parseDateTime` fix are all confirmed live bugs.

The branch conflicts with main now, so it needs a rebase. Specifics worth knowing before you do:

- **Drop** the `index.ts` AWS-region description reflow hunk — main already has it; it's the main conflict source.
- **Hand-reapply** the auto-resolution clamp in `HistoryAPI.ts` — its anchor context changed; it belongs just after the `dataDir`/`autoDiscoveryService` snapshot at the top of the values handler.
- The `state.currentConfig!` non-null assertion in the retention-migration hunk is no longer needed (config is assigned immediately above on current main) and will trip lint.
- Everything else (`data-handler.ts`, `migration-service.ts`, the aggregation-service angular block) should apply clean — those files haven't drifted.

One ask: add coverage under `test/` (unit or integration) for at least the angular re-aggregation and spatial-null fixes — the `tests/` directory isn't run by `npm test`. Main is also on eslint 10 / prettier 3.9.6 now, so run `npm run ci` after rebasing.
